librz/bin/p/bin_elf.inc

```diff
@@ -387,13 +387,235 @@ static ut64 reloc_targets_vfile_size(RzBinFile *bf, ELFOBJ *obj) {
 	return Elf_(rz_bin_elf_get_relocs_count)(obj) * reloc_target_size(obj);
 }
 
-static void patch_reloc(struct Elf_(rz_bin_elf_obj_t) * obj, RzBinElfReloc *rel, ut64 S, ut64 B, ut64 L) {
+static void patch_reloc_hexagon(RZ_INOUT RzBuffer *buf_patched, const int rel_type, const ut64 patch_addr, const ut64 S, const ut64 B, const ut64 L, const ut64 GOT, const ut64 A, const ut64 P) {
```

Comment:
Instead of passing so many parameters I suggest just create some context structure to group them into one or a few structures and pass them instead.